pkg/cmd/server/admin/create_signercert.go

```diff
@@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
 		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
 	}
 	if written {
-		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
+		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
```

Comment:
----- Original Message -----
> > @@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert()
> > (*crypto.CA, error) {
> >  		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile,
> >  		o.Name)
> >  	}
> >  	if written {
> > -		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key
> > in %s\n", o.Name, o.CertFile, o.KeyFile)
> > +		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n",
> > o.Name, o.CertFile, o.KeyFile)
> 
> > The problem is putting human messages in code reused in different contexts.
> > The general solution to that is to print messages above the shared code
> > and make sure you respond correctly.  Filtering output is focusing on
> > structuring return values out of unstructured content, which usually is a
> > mistake
> 
> As I mentioned, "That can be handled by not having a library type method
> perform output, but that seems heavy in this case".  The amount depth of
> chaining is significant, I don't envy the person doing it.

I think the Bulk pattern is the right level for this.  Not going to implement it here, but it's what we do when we have this problem in other places in the CLI.

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/2835/files#r31817725